io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java

```diff
@@ -228,7 +228,8 @@ public WfsBackedGmlInstanceCollection(LocatableInputSupplier<? extends InputStre
 		// Use primordial URI and issue "hits" request to check if the WFS will
 		// return anything at all
 		int hits;
-		if (ignoreNumberMatched) {
+
+		if (ignoreNumberMatched || primordialQueryParams.containsKey("RESOLVEDEPTH")) {
```

Comment:
Maybe add a comment why this is done in case of `RESOLVEDEPTH` being included?

Did you test if the paging works as expected when additionalObjects are returned?
I could imagine a possible impact could be that the request for the second page uses a wrong offset.

Reply:
> Did you test if the paging works as expected when additionalObjects are returned?
> I could imagine a possible impact could be that the request for the second page uses a wrong offset.

For this I would probably need a bigger project I believe. I will ask the service team.